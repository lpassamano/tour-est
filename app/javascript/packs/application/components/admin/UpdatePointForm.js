import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import PointFormFields from "./PointFormFields";
import * as pointActions from "../../redux/points/actions";
import * as pointSelectors from "../../redux/points/selectors";

export class UpdatePointForm extends Component {
  handleSubmit = attributes => {
    this.props.onUpdatePoint(this.props.tourId, { point: attributes });
  };

  render() {
    return (
      <PointFormFields
        onSubmit={this.handleSubmit}
        onHide={this.props.onHide}
        initialValues={this.props.point}
      />
    );
  }
}

UpdatePointForm.propTypes = {
  onHide: PropTypes.func.isRequired,
  onUpdatePoint: PropTypes.func.isRequired,
  tourId: PropTypes.string.isRequired,
  point: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string,
    caption: PropTypes.string,
    description: PropTypes.string,
    location: PropTypes.string,
    directions: PropTypes.string,
    image: PropTypes.string
  }).isRequired
};

const mapStateToProps = (state, ownProps) => ({
  point: pointSelectors.getPoint(state, ownProps.pointId)
});

const mapDispatchToProps = { onUpdatePoint: pointActions.updatePoint };

const enhance = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default enhance(UpdatePointForm);
