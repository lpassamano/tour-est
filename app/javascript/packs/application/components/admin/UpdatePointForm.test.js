import { UpdatePointForm } from "./UpdatePointForm";
import React from "react";
import { shallow } from "enzyme";

describe("<UpdatePointForm />", () => {
  const setup = () => {
    const point = { title: "Point 1", id: 1 };
    const onUpdatePoint = jest.fn();
    const onHide = jest.fn();
    const tourId = "1";
    const component = shallow(
      <UpdatePointForm
        point={point}
        onUpdatePoint={onUpdatePoint}
        onHide={onHide}
        tourId={tourId}
      />
    );
  };

  test("render", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
});
