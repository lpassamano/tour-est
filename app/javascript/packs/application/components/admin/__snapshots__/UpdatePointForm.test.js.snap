// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdatePointForm /> render 1`] = `undefined`;
